lumigator/python/mzai/backend/backend/tests/api/routes/test_e2e.py

```diff
@@ -0,0 +1,48 @@
+from pathlib import Path
+
+import requests
+from fastapi import status
+from fastapi.testclient import TestClient
+from schemas.datasets import DatasetFormat, DatasetResponse
+
+CWD = Path(__file__).parent
+
+
+def test_e2e_launched_backend():
```

Comment:
let's set up these tests using FastAPI's `TestClient` which does a lot of the heavy lifting for you
https://fastapi.tiangolo.com/tutorial/testing/#using-testclient

Reply:
That's what we're doing for test_dataset et al. and the dummy tests. Here I was going for the simplicity of not creating a new TestClient - but what I can do is see if we can reuse the other one.

Reply:
And upon second thought, also to keep it consistent with the rest of the codebase, we're now using a TestClient that talks to the real backend on localhost.